src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/resource/SyncAndMigrateResourceTest.kt

```diff
@@ -0,0 +1,264 @@
+      val request = MigrateRequest(
+        firstPrisonerNumber = "C7777XX",
+        firstPrisonerReason = NonAssociationReason.VICTIM,
+        secondPrisonerNumber = "D7777XX",
+        secondPrisonerReason = NonAssociationReason.PERPETRATOR,
+        restrictionType = NonAssociationRestrictionType.CELL,
+        comment = "They keep fighting",
+        authorisedBy = "Me",
+      )
+
+      // correct role, missing write scope
+      webTestClient.post()
+        .uri(url)
+        .headers(setAuthorisation(roles = listOf("ROLE_DUMMY")))
+        .header("Content-Type", "application/json")
+        .bodyValue(jsonString(request))
+        .exchange()
+        .expectStatus()
+        .isForbidden
+    }
+
+    @Test
+    fun `without a valid request body responds 400 Bad Request`() {
+      // no request body
+      webTestClient.post()
+        .uri(url)
+        .headers(
+          setAuthorisation(
+            roles = listOf("ROLE_NON_ASSOCIATIONS_MIGRATE"),
+          ),
+        )
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus()
+        .isBadRequest
+
+      // unsupported Content-Type
+      webTestClient.post()
+        .uri(url)
+        .headers(
+          setAuthorisation(
+            roles = listOf("ROLE_NON_ASSOCIATIONS_MIGRATE"),
+          ),
+        )
+        .header("Content-Type", "text/plain")
+        .bodyValue("{}")
+        .exchange()
+        .expectStatus()
+        .isBadRequest
+
+      // request body missing some fields
+      webTestClient.post()
+        .uri(url)
+        .headers(
+          setAuthorisation(
+            roles = listOf("ROLE_NON_ASSOCIATIONS_MIGRATE"),
+          ),
+        )
+        .header("Content-Type", "application/json")
+        .bodyValue(jsonString("firstPrisonerNumber" to "A1234BC"))
+        .exchange()
+        .expectStatus()
+        .isBadRequest
+    }
+
+    @Test
+    fun `for a valid request migrates the non-association`() {
+      val request = MigrateRequest(
+        firstPrisonerNumber = "C7777XX",
+        firstPrisonerReason = NonAssociationReason.VICTIM,
+        secondPrisonerNumber = "D7777XX",
+        secondPrisonerReason = NonAssociationReason.PERPETRATOR,
+        restrictionType = NonAssociationRestrictionType.CELL,
+        comment = "This is a comment",
+      )
+
+      val expectedResponse =
+        // language=json
+        """
+        {
+          "firstPrisonerNumber": "${request.firstPrisonerNumber}",
+          "firstPrisonerReason": "${request.firstPrisonerReason}",
+          "secondPrisonerNumber": "${request.secondPrisonerNumber}",
+          "secondPrisonerReason": "${request.secondPrisonerReason}",
+          "restrictionType": "${request.restrictionType}",
+          "comment": "${request.comment}",
+          "authorisedBy": "$SYSTEM_USERNAME",
+          "isClosed": false,
+          "closedReason": null,
+          "closedBy": null,
+          "closedAt": null
```

Comment:
Is the assumption here that all non-associations migrated via this endpoint will be open? Aren't we going to migrate existing closed non-associations too?

Reply:
yes we are